server/src/main/java/org/elasticsearch/indices/recovery/RecoveryTarget.java

```diff
@@ -522,6 +525,59 @@ public void writeFileChunk(StoreFileMetaData fileMetaData, long position, BytesR
         }
     }
 
+    @Override
+    public CompletableFuture<Void> writeFileChunk(StoreFileMetaData fileMetaData, long position, BytesReference content,
```

Comment:
CompletableFuture has problems with swallowing Errors (see https://github.com/elastic/elasticsearch/pull/32512). Let's use `ListenableFuture` for now.

Reply:
done